nautobot/dcim/forms.py

```diff
@@ -3625,12 +3644,12 @@ def clean(self):
         super().clean()
 
         # Untagged interfaces cannot be assigned tagged VLANs
-        if self.cleaned_data["mode"] == InterfaceModeChoices.MODE_ACCESS and self.cleaned_data["tagged_vlans"]:
+        if self.cleaned_data["mode"] == InterfaceModeChoices.MODE_ACCESS and self.cleaned_data["add_tagged_vlans"]:
             raise forms.ValidationError({"mode": "An access interface cannot have tagged VLANs assigned."})
 
         # Remove all tagged VLAN assignments from "tagged all" interfaces
         elif self.cleaned_data["mode"] == InterfaceModeChoices.MODE_TAGGED_ALL:
-            self.cleaned_data["tagged_vlans"] = []
+            self.cleaned_data["remove_tagged_vlans"] = []
```

Comment:
This change seems wrong to me? I'd think that we'd need to set "remove_tagged_vlans" to `VLAN.objects.all()` to keep this working, which seems undesirable at scale in any case. Might be an argument for keeping the direct `tagged_vlans` assignment field in parallel.